File: src/Application Specific/Classes/Entity Collections/Spares/Product/ProductValidator.vb
Namespace Entity

Namespace Products

Public Class ProductValidator
    'make sure that contact object is valid
    Inherits BaseValidator

    Public Sub Validate(ByVal oProduct As Product)

        'make sure that contact object is valid
        If oProduct.Errors.Count > 0 Then
            Dim de As DictionaryEntry
            For Each de In oProduct.Errors
                        If de.Key = "RecommendedQuantity" Then
                            Me.AddCriticalMessage("Maximum Quantity Invalid")
                        ElseIf de.Key = "MinimumQuantity" Then
                            Me.AddCriticalMessage("Minimum Quantity Invalid")
                        Else
                            Me.AddCriticalMessage(CStr(de.Value))
                        End If
            Next
        End If

                If oProduct.Number.Trim.Length = 0 Then
                    Me.AddCriticalMessage("*GC Number Missing")
                Else
                    If DB.Product.Check_Unique_Number(oProduct.Number, oProduct.ProductID) = False Then
                        Me.AddCriticalMessage("GC Number already exists")
                    End If
                End If
                If oProduct.TypeID = 0 Then
                    Me.AddCriticalMessage("*Type Missing")
                End If
                If oProduct.MakeID = 0 Then
                    Me.AddCriticalMessage("*Make Missing")
                End If
                If oProduct.ModelID = 0 Then
                    Me.AddCriticalMessage("*Model Missing")
                End If
           
                If oProduct.RecommendedQuantity < oProduct.MinimumQuantity Then
                    Me.AddCriticalMessage("*Maximum Quantity cannot be less than the Minimum Quantity")
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

    End Sub

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Spares/Product/ProductValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace Products
    {
        // make sure that contact object is valid
        public class ProductValidator : BaseValidator
        {
            public void Validate(Product oProduct)
            {

                // make sure that contact object is valid
                if (oProduct.Errors.Count > 0)
                {
                    DictionaryEntry de;
                    foreach (var de in oProduct.Errors)
                    {
                        if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(de.Key, "RecommendedQuantity", false)))
                        {
                            AddCriticalMessage("Maximum Quantity Invalid");
                        }
                        else if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(de.Key, "MinimumQuantity", false)))
                        {
                            AddCriticalMessage("Minimum Quantity Invalid");
                        }
                        else
                        {
                            AddCriticalMessage(Conversions.ToString(de.Value));
                        }
                    }
                }

                if (oProduct.Number.Trim().Length == 0)
                {
                    AddCriticalMessage("*GC Number Missing");
                }
                else if (App.DB.Product.Check_Unique_Number(oProduct.Number, oProduct.ProductID) == false)
                {
                    AddCriticalMessage("GC Number already exists");
                }

                if (oProduct.TypeID == 0)
                {
                    AddCriticalMessage("*Type Missing");
                }

                if (oProduct.MakeID == 0)
                {
                    AddCriticalMessage("*Make Missing");
                }

                if (oProduct.ModelID == 0)
                {
                    AddCriticalMessage("*Model Missing");
                }

                if (oProduct.RecommendedQuantity < oProduct.MinimumQuantity)
                {
                    AddCriticalMessage("*Maximum Quantity cannot be less than the Minimum Quantity");
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}